hack/build-image/Dockerfile

```diff
@@ -12,40 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.12
+FROM golang:1.13
 
+# Use a proxy for go modules to reduce the likelihood of various hosts being down and breaking the build
+ENV GOPROXY=https://proxy.golang.org
+
+# get code-generation tools (for now keep in GOPATH since they're not fully modules-compatible yet)
 RUN mkdir -p /go/src/k8s.io
 WORKDIR /go/src/k8s.io
 RUN git config --global advice.detachedHead false
 RUN git clone -b kubernetes-1.17.0 https://github.com/kubernetes/code-generator
-RUN git clone -b kubernetes-1.17.0 https://github.com/kubernetes/apimachinery 
-# Use a proxy for go modules to reduce the likelihood of various hosts being down and breaking the build
-ENV GOPROXY=https://proxy.golang.org
 # vendor code-generator go modules to be compatible with pre-1.15
-WORKDIR  /go/src/k8s.io/code-generator
+WORKDIR /go/src/k8s.io/code-generator
 # Don't use ENV here because we don't want to disable modules for subsequent commands
 RUN GO111MODULE=on go mod vendor
-RUN mkdir -p /go/src/sigs.k8s.io
-WORKDIR /go/src/sigs.k8s.io
-RUN git clone -b v0.2.2 https://github.com/kubernetes-sigs/controller-tools
-WORKDIR /go/src/sigs.k8s.io/controller-tools
-RUN GO111MODULE=on go mod vendor
-RUN go get golang.org/x/tools/cmd/goimports
-WORKDIR /go/src/golang.org/x/tools
-RUN git checkout 40a48ad93fbe707101afb2099b738471f70594ec
-RUN go install ./cmd/goimports
-RUN echo chmod -R a+w /go
+
+# get controller-tools
+RUN GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.2
```

Comment:
how will we track/maintain this specific dependency?

Reply:
for now we just track it through the Dockerfile. I'm going to file a follow-up issue to look at migrating to a `tools.go` approach.